webrender_api/src/image.rs

```diff
@@ -74,17 +114,35 @@ impl ImageFormat {
     }
 }
 
+/// Metadata (but not storage) describing an image In WebRender.
 #[derive(Copy, Clone, Debug, Deserialize, PartialEq, Serialize)]
 pub struct ImageDescriptor {
+    /// Format of the image data.
     pub format: ImageFormat,
+    /// Width and length of the image data, in pixels.
     pub size: DeviceUintSize,
+    /// The number of bytes from the start of one row to the next. Lazily
+    /// computed via compute_stride() and cached here when the image is
+    /// tile.
+    ///
+    /// FIXME(bholley): The computation of this looks pretty cheap. Do we have
+    /// evidence that this is hot enough that it needs caching?
```

Comment:
This doesn't look like a cache, it's an override for when the stride is not width times bpp.

Reply:
Yep, what Emilio said. Different source of images have different alignment constraints for rows, so the stride isn't always equal to width * bpp.